sql/core/src/test/scala/org/apache/spark/sql/streaming/StreamSuite.scala

```diff
@@ -1125,6 +1126,37 @@ class StreamSuite extends StreamTest {
       }
     )
   }
+
+  // ProcessingTime trigger generates MicroBatchExecution, and ContinuousTrigger starts a
+  // ContinuousExecution
+  Seq(Trigger.ProcessingTime("1 second"), Trigger.Continuous("1 second")).foreach { trigger =>
+    test(s"SPARK-30143: stop waits until timeout if blocked - trigger: $trigger") {
+      val latch = new CountDownLatch(1)
+      BlockOnStopSourceProvider.setLatch(latch)
```

Comment:
super nit: the latch can be hidden in the implementation, behind two simple APIs, `enableBlocking` and `disableBlocking`.